task/hiffy/src/common.rs

```diff
@@ -729,6 +731,56 @@ pub(crate) fn sprot_block_size(
     Ok(4)
 }
 
+#[cfg(feature = "sprot")]
+pub(crate) fn sprot_reset_component(
+    stack: &[Option<u32>],
+    data: &[u8],
+    _rval: &mut [u8],
+) -> Result<usize, Failure> {
+    if stack.len() < 3 {
+        return Err(Failure::Fault(Fault::MissingParameters));
+    }
+    let fp = stack.len() - 3;
+    let intent: drv_update_api::ResetIntent = match stack[fp + 0] {
+        // Some(intent) => match drv_update_api::ResetIntent::try_from(intent as u8) {
+        Some(intent) => {
+            match drv_update_api::ResetIntent::from_u8(intent as u8) {
+                Some(intent) => intent,
+                None => return Err(Failure::Fault(Fault::BadParameter(0))),
+            }
+        }
+        None => return Err(Failure::Fault(Fault::EmptyParameter(0))),
+    };
+    let target: drv_update_api::UpdateTarget = match stack[fp + 1] {
+        Some(target) => match drv_update_api::UpdateTarget::from_u32(target) {
+            Some(target) => target,
+            None => return Err(Failure::Fault(Fault::BadParameter(1))),
+        },
+        None => return Err(Failure::Fault(Fault::EmptyParameter(1))),
+    };
+    let auth_len: usize = match stack[fp + 2] {
+        Some(auth_len) => auth_len as usize,
+        None => return Err(Failure::Fault(Fault::EmptyParameter(0))),
+    };
+
+    if auth_len > data.len().min(drv_update_api::AUTH_MAX_SZ) {
+        return Err(Failure::Fault(Fault::AccessOutOfBounds));
+    }
+
+    let mut auth_data = [0u8; drv_update_api::AUTH_MAX_SZ];
```

Comment:
I don't understand what this code is doing. AFAICT you are
- Copying `data` to the prefix of an array
- Then passing `data`'s length to `reset_component`, causing it to ignore the rest of the array

Isn't this equivalent to, but more expensive than, just passing `data` to `reset_component` below?